packages/react-native/src/primitives/Button/Button.tsx

```diff
@@ -1,18 +1,27 @@
 import React from 'react';
 import { Pressable, Text } from 'react-native';
 
+import { useTheme } from '../../hooks';
 import { styles } from './styles';
 import { ButtonProps } from './types';
 
 export default function Button({
   children,
+  containerStyle,
   textStyle,
   ...pressableProps
 }: ButtonProps): JSX.Element {
+  const theme = useTheme();
+  const themedButtonStyle = theme.tokens?.components.button;
```

Comment:
This could be extracted into a HOC `withTheme` - we would need to pass in a nested style prop instead of flat structure of style props